app/jobs/in_person/email_reminder_job.rb

```diff
@@ -14,28 +17,47 @@ class EmailReminderJob < ApplicationJob
     def perform(_now)
       return true unless IdentityConfig.store.in_person_proofing_enabled
 
-      # final reminder job done first in case of job failure
-      second_set_enrollments = InPersonEnrollment.needs_late_email_reminder(
+      # send late emails first in case of job failure
+      late_enrollments = InPersonEnrollment.needs_late_email_reminder(
         late_benchmark,
         final_benchmark,
       )
-      second_set_enrollments.each do |enrollment|
-        send_reminder_email(enrollment.user, enrollment)
-        enrollment.update!(late_reminder_sent: true)
-      end
+      send_emails_for_enrollments(enrollments: late_enrollments, email_type: EMAIL_TYPE_LATE)
 
-      first_set_enrollments = InPersonEnrollment.needs_early_email_reminder(
+      early_enrollments = InPersonEnrollment.needs_early_email_reminder(
         early_benchmark,
         late_benchmark,
       )
-      first_set_enrollments.each do |enrollment|
-        send_reminder_email(enrollment.user, enrollment)
-        enrollment.update!(early_reminder_sent: true)
-      end
+      send_emails_for_enrollments(enrollments: early_enrollments, email_type: EMAIL_TYPE_EARLY)
     end
 
     private
 
+    def analytics(user: AnonymousUser.new)
+      Analytics.new(user: user, request: nil, session: {}, sp: nil)
+    end
+
+    def send_emails_for_enrollments(enrollments:, email_type:)
+      enrollments.each do |enrollment|
+        begin
+          send_reminder_email(enrollment.user, enrollment)
+        rescue StandardError => err
+          NewRelic::Agent.notice_error(err)
+          analytics(user: enrollment.user).idv_in_person_email_reminder_job_exception(
+            enrollment_id: enrollment.id,
+            exception_class: err.class.to_s,
+            exception_message: err.message,
+          )
+        else
+          analytics(user: enrollment.user).idv_in_person_email_reminder_job_email_initiated(
+            email_type: email_type,
+            enrollment_id: enrollment.id,
+          )
+          enrollment.update!({ "#{email_type}_reminder_sent": true })
+        end
+      end
```

Comment:
we can use the implicit top level rescue for `do...end` blocks and unindent by a bit (I just did a local test to confirm the `else` works too)

```suggestion
      enrollments.each do |enrollment|
        send_reminder_email(enrollment.user, enrollment)
      rescue StandardError => err
        NewRelic::Agent.notice_error(err)
        analytics(user: enrollment.user).idv_in_person_email_reminder_job_exception(
          enrollment_id: enrollment.id,
          exception_class: err.class.to_s,
          exception_message: err.message,
        )
      else
        analytics(user: enrollment.user).idv_in_person_email_reminder_job_email_initiated(
          email_type: email_type,
          enrollment_id: enrollment.id,
        )
        enrollment.update!({ "#{email_type}_reminder_sent": true })
      end
```

Reply:
done in 06e485d12f6e0904fbf0ff6a85b63efb61268c34